index.html

```diff
@@ -2406,6 +2634,1425 @@ <h2>
+      </p>
+      <p>
+        A [=simulated gamepad=] is not a physical device and does not have an [=input control layout=].
+        If a [=simulated gamepad=] has device identifiers that match a [=gamepad=] with a known [=input control layout=], the [=user agent=] SHOULD process inputs on the [=simulated gamepad=] as if it had the same [=input control layout=].
+      </p>
+      <section>
+        <h3>
+          The emulation module
+        </h3>
+        <section>
+          <h4>
+            Types
+          </h4>
+          <section>
+            <h5>
+              The emulation.GamepadBounds Type
+            </h5>
+            <pre class="cddl">
+              emulation.GamepadBounds = {
+                max: float64,
+                ? min: float64,
+              }
+            </pre>
+            <dl>
+              <dt>
+                max
+              </dt>
+              <dd>
+                is the [=logical maximum=] for an [=input value=].
+              </dd>
+              <dt>
+                min
+              </dt>
+              <dd>
+                is the [=logical minimum=] for an [=input value=].
+              </dd>
+            </dl>
+          </section>
+          <section>
+            <h5>
+              The emulation.SurfaceBounds Type
+            </h5>
+            <pre class="cddl">
+              emulation.SurfaceBounds = {
+                x: emulation.GamepadBounds,
+                y: emulation.GamepadBounds,
+              }
+            </pre>
+            <dl>
+              <dt>
+                x
+              </dt>
+              <dd>
+                is the logical bounds for the [=touch x coordinate=].
+              </dd>
+              <dt>
+                y
+              </dt>
+              <dd>
+                is the logical bounds for the [=touch y coordinate=].
+              </dd>
+            </dl>
+          </section>
+          <section>
+            <h5>
+              The emulation.GamepadVibrationEffectType Type
+            </h5>
+            <dl>
+              <dt>
+                "dual-rumble"
+              </dt>
+              <dd>
+                represents the {{GamepadHapticEffectType/"dual-rumble"}} {{GamepadHapticEffectType}}.
+              </dd>
+              <dt>
+                "trigger-rumble"
+              </dt>
+              <dd>
+                represents the {{GamepadHapticEffectType/"trigger-rumble"}} {{GamepadHapticEffectType}}.
+              </dd>
+            </dl>
+            <pre class="cddl">
+              emulation.GamepadVibrationEffectType = ("dual-rumble", "trigger-rumble")
+            </pre>
+          </section>
+          <section>
+            <h5>
+              The emulation.SimulateGamepadParams Type
+            </h5>
+            <pre class="cddl">
+              emulation.SimulateGamepadParams = {
+                type: "gamepad",
+                context: browsingContext.BrowsingContext,
```

Comment:
IDK if we can cross-reference CDDL to BiDi spec, so let's use `text` the same way [Bluetooth spec does](https://webbluetoothcg.github.io/web-bluetooth/#issue-194ad560).
```suggestion
                context: text,
```